scripts/setup/install.ps1

```diff
@@ -42,6 +42,48 @@ Param(
 
 $_rollbackStore = @{}
 
+function Retry-Command
+{
+    param (
+    [Parameter(Mandatory=$true)][string]$command,
+    [Parameter(Mandatory=$true)][hashtable]$args,
+    [Parameter(Mandatory=$false)][int]$retries = 2,
+    [Parameter(Mandatory=$false)][int]$secondsDelay = 1,
+    [Parameter(Mandatory=$false)][switch]$throwError
+    )
+
+    # Setting ErrorAction to Stop is important. This ensures any errors that occur in the command are
+    # treated as terminating errors, and will be caught by the catch block.
+    $args.ErrorAction = "Stop"
+
+    $retrycount = 0
+    $completed = $false
+
+    while (-not $completed) {
+        try {
+            & $command @args
+            Write-Verbose ("Command {0} succeeded." -f $command)
+            $completed = $true
+
+        } catch {
+            if ($retrycount -ge $retries) {
+                Write-Warning ("Command {0} failed the maximum number of {1} times. throwError:{2}" -f $command, $retrycount, $throwError)
+                if ($throwError) {
+                    throw
+                }
+                else
+                {
+                    break;
+                }
+            } else {
+                Write-Verbose ("Command {0} failed. Retrying in {1} seconds." -f $command, $secondsDelay)
```

Comment:
We should always put `$_.Exception` in our error AND verbose message

Reply:
Good point. Done. I used $_.Exception.Message like other functions do.